### PR TITLE
Show a way to search even when page scrolled

### LINKED ISSUE #44561: Docs search gets hidden on scroll down in page
**This is a Feature Request**

**What would you like to be added**
Currently Docs search hides when you scroll down the page, but it should not get hidden it must stay there at the top.

**Why is this needed**

- Search Input should always stay at its place.

- Also, as a suggestion it can also be called on keyboard shortcut like `ctrl + /`  like this to easily navigation in the docs.

**Comments**

Screenshot 2023-12-28 203248

### PR DESCRIPTION
**Pull request by coder12git**
fixes #44561

Check this [comment](https://github.com/kubernetes/website/issues/44561#issuecomment-1871322494)

### CODE REVIEW

**Comment by coder12git** [comment]
Screenshot 2023-12-29 202034

Screenshot 2023-12-29 202015

File: layouts/partials/navbar.html [hunk 1]
```diff
@@ -24,6 +24,9 @@
 				{{ partial "navbar-lang-selector.html" . }}
 			</li>
 			{{ end }}
+			<li class="search-item nav-item mr-n4 mr-lg-0">
+				{{ partial "search-input.html" . }}
+			</li>
```
Comment (sftim) [inline layouts/partials/navbar.html:29]: We might want to show this only on wide-enough viewports.

**Review by sftim — COMMENTED** [review]
Thanks

I would add more CSS / SCSS rules to suppress the search for narrow-enough viewports.

Separately, a more broad redesign might be useful - I really would like us to align more closely with upstream Docsy, or with https://github.com/cncf/dot-org-hugo-theme.
I'd like us to be wary of making too many changes that add to the long-term burden of site maintenance.

Even so, this is a useful change to make, and with the viewport size thing addressed, I'd support merging it.

File: layouts/partials/navbar.html (hunk 1 above)
Comment (coder12git) [reply layouts/partials/navbar.html:29]: Done

File: assets/scss/_custom.scss [hunk 2]
```diff
@@ -239,6 +239,19 @@ body.td-404 main .error-details {
...
 }
 
+.flip-nav .search-item {
+  .td-search-input {
+    background-color: $dark-grey;
+  }
+}
+
+@media only screen and (max-width: 1500px) {
+  .search-item {
```
Comment (sftim) [inline assets/scss/_custom.scss:249]: I'd be specific:
```suggestion
  header nav .search-item {
```

**Comment by sftim** [comment]
/retitle Show a way to search even when page scrolled

**Review by sftim — COMMENTED** [review]
/lgtm

This PR does what it says in its description

**Comment by sftim** [comment]
Thank you @coder12git.

I'm neutral on whether we approve this. Happy to see it land; happy to see a different answer to the same problem.

**Review by Gauravpadam — COMMENTED** [review]
Thanks, I like the idea of search on top nav

However, you might need to change the font color (or background color) of the search bar on scrolling down; The input is camouflaged atm

**Comment by coder12git** [comment]
> Thanks, I like the idea of search on top nav
> 
> However, you might need to change the font color (or background color) of the search bar on scrolling down; The input is camouflaged atm

i tried all the background-colors, i think black color is good here, as it's directly taken from color-theme

**Comment by coder12git** [comment]
@Gauravpadam plz check now, and let me know if it's fine or not?

**Comment by Gauravpadam** [comment]
Thanks

This lgtm; +1 for https://github.com/kubernetes/website/pull/44574#issuecomment-1872253804

**Comment by coder12git** [comment]
@sftim any feedback here?

**Comment by sftim** [comment]
I'm [still](https://github.com/kubernetes/website/pull/44574#issuecomment-1872253804) neutral on whether we approve this. Happy to see it land; happy to see a different answer to the same problem.

/lgtm

**Comment by dipesh-rawat** [comment]
> If this PR gets five 👍 reactions and no 👎 reactions, let's take that as ground for approval even if we have no other grounds. If we find another reason to accept it, that is good too.

Based on the criteria [outlined](https://github.com/kubernetes/website/pull/44574#issuecomment-1941975096) above for merging, it appears that we've achieved a consensus. I'm happy to see this get merged.

/approve

**Comment by sftim** [comment]
Thanks for this @coder12git

**Pull request merged**